accord-core/src/main/java/accord/local/CommandsForKey.java

```diff
@@ -290,6 +302,75 @@ public static InternalStatus get(int ordinal)
         }
     }
 
+    static class LoadingPruned extends TxnId
```

Comment:
This could use a comment explaining what it does, and maybe a new name. I was able to work out it's a reverse deps index, mapping a txnid to txnids that have it as a dep. However I don't see the list updated in `pruneBefore` and I'm not sure what the loading part is referring to.

Reply:
There is some doc about this, but minimal and not spread widely enough, and probably not at the class level. I will update that to describe what's going on here. This is the "once we prune we are left with potentially incomplete information, and have to sometimes load per-command information from disk" part of the patch. 

Pruned TxnId are those we have decided we probably don't need as we expect them to be transitive dependencies of transactions we've already applied, but that could still appear as dependencies again, but also might just be unwitnessed transactions with an old id - and to know which it is, we have to load the TxnId to find out what the command-level data knows of it.

Reply:
That is to say, pruning doesn't trigger this, but encountering a `TxnId` dependency that occurs before `prunedBefore` does.